packages/rocketchat-api/server/v1/settings.js

```diff
@@ -1,4 +1,35 @@
 // settings endpoints
+RocketChat.API.v1.addRoute('settings', { authRequired: true }, {
+	get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort, fields, query } = this.parseJsonQuery();
+
+		let ourQuery = {
+			hidden: { $ne: true }
+		};
+
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-privileged-setting')) {
+			ourQuery.public = { $ne: false };
```

Comment:
I prefer this comparison like `public: true` to prevent return settings no explicitly set as public.